src/realm/sync/client_base.hpp

```diff
@@ -185,6 +186,10 @@ struct ClientConfig {
     // and creating WebSockets. If not provided the default implementation will be used.
     std::shared_ptr<sync::SyncSocketProvider> socket_provider;
 
+    // Optional thread observer for event loop thread events in the default SyncSocketProvider
+    // implementation. It is not used for custom SyncSocketProvider implementations.
+    std::shared_ptr<BindingCallbackThreadObserver> default_socket_provider_thread_observer;
```

Comment:
does this really need shared lifetime management? could we just use a unique_ptr here?

Reply:
It could technically just be a unique_ptr, since the SyncClientConfig is required for each start of the sync client. The main question is whether the SDK would prefer to use a single shared ptr across all the apps or is it fine to define a separate observer instance for each app. @cmelchior, any preferences?